cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -64,8 +64,12 @@ const (
 	conflictRetryInterval   = 5 * time.Second
 	// machinePriorityAnnotation is the annotation to set machine priority while deletion
 	machinePriorityAnnotation = "machinepriority.machine.sapcloud.io"
-	// labelForFailureDomain is the label for failure domain used by kubernetes.io
-	labelForFailureDomain = "failure-domain.beta.kubernetes.io/zone"
+	// labelFailureDomainZone is the deprecated label for failure domain zone used by kubernetes.io
+	// It has been deprecated in favor of (below) topology.kubernetes.io/zone label
+	// See: https://v1-18.docs.kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#failure-domainbetakubernetesiozone
+	labelFailureDomainZone = "failure-domain.beta.kubernetes.io/zone"
```

Comment:
One minor comment, with rebase these labels now can be fetched from `apiv1 "k8s.io/api/core/v1" ` as well.